File: Programação Para Dispositivos Móveis/Exercícios em Sala App/Aula 08/ConversorTemperatura/app/src/main/java/com/pedro/conversortemperatura/MainActivity.java
package com.pedro.conversortemperatura;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        EditText textoTemperatura = findViewById(R.id.editTextTemperatura);
        Button botaoConverterCelsius = findViewById(R.id.buttonConverterParaCelsius);
        Button botaoConverterFahrenheit = findViewById(R.id.buttonConverterParaFahrenheit);
        TextView textoResultado = findViewById(R.id.textViewResultado);

        botaoConverterCelsius.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String campoTemperaturaFahrenheit = textoTemperatura.getText().toString();

                if(campoTemperaturaFahrenheit.isEmpty())
                    Toast.makeText(getApplicationContext(), "Preencha o campo de temperatura",Toast.LENGTH_SHORT).show();
                else {
                    double temperatura = Double.parseDouble(campoTemperaturaFahrenheit);
                    double celsius = (temperatura-32)*5/9;
                    textoResultado.setText(String.format("Temperatura: %.2f",celsius));
                }
            }
        });

        botaoConverterFahrenheit.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String campoTemperaturaCelsius = textoTemperatura.getText().toString();

                if(campoTemperaturaCelsius.isEmpty())
                    Toast.makeText(getApplicationContext(), "Preencha o campo de temperatura",Toast.LENGTH_SHORT).show();
                else {
                    double temperatura = Double.parseDouble(campoTemperaturaCelsius);
                    double fahrenheit = (temperatura*9/5)+32;
                    textoResultado.setText(String.format("Temperatura: %.2f", fahrenheit));
                }
            }
        });
    }
}

File: Programação Para Dispositivos Móveis/Exercícios em Sala App/Aula 09/AcessandoAPI/app/src/main/java/com/example/acessandoapi/MainActivity.kt
package com.example.acessandoapi

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class MainActivity : AppCompatActivity() {

    private lateinit var editTextCep: EditText
    private lateinit var buttonBuscar: Button
    private lateinit var textViewResultado: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        editTextCep = findViewById(R.id.editTextCep)
        buttonBuscar = findViewById(R.id.buttonBuscar)
        textViewResultado = findViewById(R.id.textViewResultado)

        buttonBuscar.setOnClickListener {
            val cep = editTextCep.text.toString()

            if (cep.isNotEmpty()) {
                buscarCep(cep)
            } else {
                Toast.makeText(this, "Por favor, digite um CEP", Toast.LENGTH_SHORT).show()
            }
        }
    }

    private fun buscarCep(cep: String) {
        val call = RetrofitClient.instance.buscarCep(cep)

        call.enqueue(object : Callback<CepResponse> {

            override fun onResponse(call: Call<CepResponse>, response: Response<CepResponse>) {
                if (response.isSuccessful) {
                    val cepResponse = response.body()
                    cepResponse?.let {
                        val resultado = """
                        CEP: ${it.cep}
                        Logradouro: ${it.logradouro}
                        Bairro: ${it.bairro}
                        Localidade: ${it.localidade}
                        UF: ${it.uf}
                    """.trimIndent()
                        textViewResultado.text = resultado
                    }
                } else {
                    Toast.makeText(this@MainActivity, "CEP não encontrado!", Toast.LENGTH_SHORT).show()
                }
            }

            override fun onFailure(call: Call<CepResponse>, t: Throwable) {
                Toast.makeText(this@MainActivity, "Erro ao buscar o CEP: ${t.message}", Toast.LENGTH_SHORT).show()
            }
        })
    }
}
